static/js/components/ObjectTags.jsx

```diff
@@ -0,0 +1,331 @@
+  const handleTagChange = (event) => {
+    setSelectedTagId(event.target.value);
+  };
+
+  const refreshSource = () => {
+    if (source && source.id) {
+      dispatch(sourceActions.fetchSource(source.id));
+    }
+  };
+
+  const handleNewTagNameChange = (event) => {
+    setNewTagName(event.target.value);
+  };
+
+  const handleCreateTag = () => {
+    if (!newTagName.trim()) {
+      setTagError("Tag name cannot be empty");
+      return;
+    }
+
+    setIsCreatingTag(true);
+
+    dispatch(
+      objectTagsActions.createTagOption({
+        name: newTagName,
+      }),
+    ).then((result) => {
+      setIsCreatingTag(false);
+
+      if (result.status === "success") {
+        dispatch(showNotification("Tag created successfully"));
+        dispatch(objectTagsActions.fetchTagOptions());
+        setNewTagName("");
+
+        if (result.data && result.data.id) {
+          setSelectedTagId(result.data.id);
+        }
+      } else {
+        const errorMsg = result.message || "Failed to create tag";
+        setTagError(errorMsg);
+        dispatch(showNotification(errorMsg, "error"));
+      }
+    });
+  };
+
+  const handleAddTag = () => {
+    if (!selectedTagId) return;
+
+    setIsAddingTag(true);
+
+    dispatch(
+      objectTagsActions.addObjectTag({
+        obj_id: source.id,
+        objtagoption_id: selectedTagId,
+      }),
+    ).then((result) => {
+      setIsAddingTag(false);
+      if (result.status === "success") {
+        dispatch(showNotification("Tag added successfully"));
+
+        refreshSource();
+
+        handleCloseDialog();
+      } else {
+        dispatch(showNotification("Failed to add tag", "error"));
+      }
+    });
+  };
+
+  const handleDeleteTag = (association_id) => {
+    dispatch(objectTagsActions.deleteObjectTag({ id: association_id })).then(
+      (result) => {
+        if (result.status === "success") {
+          dispatch(showNotification("Source Tag deleted"));
+
+          refreshSource();
+        } else {
+          dispatch(showNotification("Failed to delete tag", "error"));
+        }
+      },
+    );
+  };
+
+  const usedTagIds = (source.tags || []).map((tag) => tag.objtagoption_id);
+  const availableTags = tagOptions.filter(
+    (tag) => !usedTagIds.includes(tag.id),
+  );
+
+  return (
+    <div className={styles.root}>
+      <div className={styles.chips}>
+        {(source.tags || []).map((tag) => (
+          <Chip
+            className={styles.chip}
+            key={tag.id}
+            label={tag.name}
+            size="small"
+            onDelete={() => handleDeleteTag(tag.id)}
+            data-testid={`tag-chip-${tag.id}`}
+          />
+        ))}
+      </div>
+
+      <Tooltip title="Add Tag">
+        <IconButton
+          size="small"
+          className={styles.addButton}
+          onClick={handleOpenDialog}
+          data-testid="add-tag-button"
+        >
+          <AddIcon fontSize="small" />
+        </IconButton>
+      </Tooltip>
+
+      <Dialog
```

Comment:
We usually avoid using custom forms when we can, to stick with forms from react json schema form. Any reason to use a custom form here? Not saying it's incorrect, just curious.